maintainers/maintainer-list.nix

```diff
@@ -13144,6 +13144,12 @@
     githubId = 10420834;
     name = "Mihai-Drosi Caju";
   };
+  mccartykim = {
```

Comment:
Split this in a separate commit (no need for opening another PR):

`maintainers: add mccartykim`